packages/ember-glimmer/lib/setup-registry.js

```diff
@@ -14,15 +14,24 @@ export function setupApplicationRegistry(registry) {
   registry.register('renderer:-dom', InteractiveRenderer);
   registry.register('renderer:-inert', InertRenderer);
 
-  let { DOMChanges, DOMTreeConstruction } = require('ember-glimmer/dom');
+  let { DOMChanges, DOMTreeConstruction, NodeDOMTreeConstruction } = require('ember-glimmer/dom');
+
 
   registry.register('service:-dom-changes', {
     create({ document }) { return new DOMChanges(document); }
   });
 
-  registry.register('service:-dom-tree-construction', {
-    create({ document }) { return new DOMTreeConstruction(document); }
-  });
+  if (typeof document !== 'undefined') {
+    registry.register('service:-dom-tree-construction', {
+      create({ document }) { return new DOMTreeConstruction(document); }
+    });
+
+  } else {
+    registry.register('service:-dom-tree-construction', {
+      create({ document: simpleDocument }) { return new NodeDOMTreeConstruction(simpleDocument); }
```

Comment:
This param is the injection setup ([from here](https://github.com/emberjs/ember.js/blob/master/packages/ember-application/lib/system/application-instance.js#L289)), and shouldn't need to be overridden like this.

I suspect something like this would work (once the `service:-document` injection is fixed):

```js
registry.register('service:-dom-tree-construction', {
  create({ document }) {
    let Implementation = environment.hasDOM ? DOMTreeConstruction : NodeDOMTreeConstruction;
    return new Implementation(document);
  }
});
```